lib/logstash/config/config_ast.rb

```diff
@@ -211,13 +216,7 @@ def compile
         return "start_input(#{variable_name})"
       when "filter"
         return <<-CODE
-          events = events.flat_map do |event|
-            next [] if event.cancelled?
-
-            new_events = []
-            #{variable_name}.filter(event){|new_event| new_events << new_event}
-            event.cancelled? ? new_events : new_events.unshift(event)
-          end
+          events = #{variable_name}.multi_filter(events)
         CODE
```

Comment:
no, plugins always yield only one event, **this has not changed** and we do not want to change any API at this point. The proposal for the filter method refactor is in #2872 and we'll do that in 1.6 or 2.0.

the previous generated code, to handle new events emitted by the filter was inserting these new events as an array in the events array, see generated code example in section **1.5.0-rc2** of https://github.com/elastic/logstash/pull/2869#issuecomment-85132637 
now, the `multi_filter` method takes care of that logic but adds individual events not as an array, so `flat_map` is not required anymore.